lib/ansible/modules/cloud/google/gcp_target_proxy.py

```diff
@@ -0,0 +1,316 @@
+    required: true
+  url_map:
+    description:
+       - Name of the Url Map.  Required if type is HTTP or HTTPS proxy.
+    required: false
+'''
+
+EXAMPLES = '''
+- name: Create Minimum HTTP Target_Proxy
+  gcp_target_proxy:
+    service_account_email: "{{ service_account_email }}"
+    credentials_file: "{{ credentials_file }}"
+    project_id: "{{ project_id }}"
+    target_proxy_name: my-target_proxy
+    target_proxy_type: HTTP
+    url_map: my-url-map
+    state: present
+'''
+
+RETURN = '''
+state:
+    description: state of the Target_Proxy
+    returned: Always.
+    type: str
+    sample: present
+updated_target_proxy:
+    description: True if the target_proxy has been updated. Will not appear on
+                 initial target_proxy creation.
+    returned: if the target_proxy has been updated.
+    type: bool
+    sample: true
+target_proxy_name:
+    description: Name of the Target_Proxy
+    returned: Always
+    type: str
+    sample: my-target-proxy
+target_proxy_type:
+    description: Type of Target_Proxy. One of HTTP, HTTPS or SSL.
+    returned: Always
+    type: str
+    sample: HTTP
+target_proxy:
+    description: GCP Target_Proxy dictionary
+    returned: Always. Refer to GCP documentation for detailed field descriptions.
+    type: dict
+    sample: { "name": "my-target-proxy", "urlMap": "..." }
+'''
+
+
+try:
+    from ast import literal_eval
+    HAS_PYTHON26 = True
+except ImportError:
+    HAS_PYTHON26 = False
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.gcp import get_google_api_client, GCPUtils
+
+USER_AGENT_PRODUCT = 'ansible-target_proxy'
+USER_AGENT_VERSION = '0.0.1'
+
+
+def _build_target_proxy_dict(params, project_id=None):
+    """
+    Reformat services in Ansible Params.
+
+    :param params: Params from AnsibleModule object
+    :type params: ``dict``
+
+    :param project_id: The GCP project ID.
+    :type project_id:  ``str``
+
+    :return: dictionary suitable for submission to GCP UrlMap API.
+    :rtype ``dict``
+    """
+    url = ''
+    if project_id:
+        url = GCPUtils.build_googleapi_url(project_id)
+    gcp_dict = GCPUtils.params_to_gcp_dict(params, 'target_proxy_name')
+    if 'urlMap' in gcp_dict:
+        gcp_dict['urlMap'] = '%s/global/urlMaps/%s' % (url,
+                                                       gcp_dict['urlMap'])
+    return gcp_dict
+
+
+def get_target_http_proxy(client, name, project_id=None):
+    """
+    Get a Target HTTP Proxy from GCP.
+
+    :param client: An initialized GCE Compute Disovery resource.
+    :type client:  :class: `googleapiclient.discovery.Resource`
+
+    :param name: Name of the Target Proxy.
+    :type name:  ``str``
+
+    :param project_id: The GCP project ID.
+    :type project_id:  ``str``
+
+    :return: A dict resp from the respective GCP 'get' request.
+    :rtype: ``dict``
+    """
+    try:
+        req = client.targetHttpProxies().get(project=project_id,
```

Comment:
If you're just raising, you can skip the try/except since it's handled in the caller.